cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -99,35 +106,75 @@ func (sub CosmosSub) Start(completionEvent *sync.WaitGroup) {
 	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
 	defer close(quit)
 
+	var lastProcessedBlock int64
+
+	data, err := sub.DB.Get([]byte(cosmosLevelDBKey), nil)
+	if err != nil {
+		log.Println("Error getting the last cosmos block from level db", err)
+		lastProcessedBlock = 0
+	} else {
+		lastProcessedBlock = new(big.Int).SetBytes(data).Int64()
+	}
+
 	for {
 		select {
-		case result := <-out:
-			tx, ok := result.Data.(tmTypes.EventDataTx)
+		case <-quit:
+			log.Println("we receive the quit signal and exit")
+			return
+
+		case e := <-results:
+			data, ok := e.Data.(tmTypes.EventDataNewBlock)
 			if !ok {
 				sub.SugaredLogger.Errorw("sifchain client failed to extract event data from new tx.",
 					errorMessageKey, err.Error())
```

Comment:
err is defined somewhere else, so instead of having this logger spit out this error message that will likely be nil. Let's remove line 129. On line 128, let's have a message that says `sifchain client failed to extract event data from new block."` and then prints out the value in `tmTypes.EventDataNewBlock` so that we know why there was a failure and we can add failsafes in if this ever happens.

Reply:
Accepted, will fix the issue.